src/client.c

```diff
@@ -1377,6 +1377,17 @@ static gchar *copy_job_as_template(gchar *job, gboolean novalid,
                                            recipe_set_node_ptr, wboard,
                                            role, owner, family);
 
+            // Copy recipe params
+            xmlXPathObjectPtr params_node = get_node_set(template_xml_doc_ptr,
+                    node, (xmlChar*)"params");
+            if (params_node) {
+                xmlNodePtr copy_params_node_ptr = xmlDocCopyNode(params_node->nodesetval->nodeTab[0],
+                                                  new_xml_doc_ptr, 1);
+                xmlAddChild (new_recipe_ptr, copy_params_node_ptr);
+            }
+            xmlXPathFreeObject(params_node);
+            params_node = NULL;
+
```

Comment:
Although xmlXPathFreeObject itself checks for NULL, the free should be inside the block,
```suggestion
                xmlNodePtr copy_params_node_ptr = xmlDocCopyNode(params_node->nodesetval->nodeTab[0],
                                                  new_xml_doc_ptr, 1);
                xmlAddChild (new_recipe_ptr, copy_params_node_ptr);
                
                xmlXPathFreeObject (params_node);
                params_node = NULL;
            }
```

Reply:
I thought the same thing but I was following the rest of the code that is making this call outside of the blocks.

I will change it though since I am probably the one who did it wrong the first time :-)

Reply:
Thanks @p3ck. Not really an issue because `xmlXPathFreeObject` checks for NULL, but I had to check [the source code in libxml2](https://gitlab.gnome.org/GNOME/libxml2/-/blob/master/xpath.c#L5459) to make sure, as documentation for the function doesn't mention it.